modules/peers.js

```diff
@@ -304,139 +383,228 @@ Peers.prototype.acceptable = function (peers) {
+	});
+};
+
+__private.dbLoad = function (cb) {
+	library.logger.trace('Importing peers from database');
+	library.db.any(sql.getAll).then(function (rows) {
+		library.logger.trace('Imported peers from database', {count: rows.length});
+		steed.each (rows, function (row, eachCb) {
+			// Delete dapp if null
+			if (!row.dappid) {
+				delete row.dappid;
+			}
+			__private.upsertPeer(row);
+			setImmediate(eachCb);
+		}, function (err) {
+			return setImmediate(cb, err);
+		});
+	}).catch(function (err) {
+        library.logger.error('Import peers from database failed', {error: err.message || err});
+        return setImmediate(cb);
+	});
+};
+
+__private.dbSave = function (cb) {
+	var peers = [];
+	// Preparing peers list (for consistency)
+	_.each(__private.peers, function (peer) {
+		peers.push(peer);
+	});
+
+	// Do nothing when peers list is empty
+	if (!peers.length) {
+		library.logger.debug('Export peers to database failed: Peers list empty');
+		return setImmediate(cb);
+	}
+
+	// Creating set of columns
+	var cs = new pgp.helpers.ColumnSet([
+		'ip', 'port',
+		{name: 'state',     def: 1},
+		{name: 'height',    def: 1},
+		{name: 'os',        def: null},
+		{name: 'version',   def: null},
+		{name: 'broadhash', def: null, init: function (col) {
+			return col.value ? new Buffer(col.value, 'hex') : null;
+		}},
+		{name: 'clock',     def: null}
+	], {table: 'peers'});
+	// Generating insert query
+	var insert_peers = pgp.helpers.insert(peers, cs);
+
+	// Wrap sql queries in transaction and execute
+	library.db.tx(function (t) {
+		var queries = [
+        	// Clear peers table
+            t.none(sql.clear),
+            // Insert all peers
+            t.none(insert_peers)
+		];
+
+		// Inserting dapps peers
+		_.each(peers, function (peer) {
+			if (peer.dappid) {
+				// If there are dapps on peer - push separately for every dapp
+				_.each (peer.dappid, function (dappid) {
+					var dapp_peer = peer;
+					dapp_peer.dappid = dappid;
+					queries.push(t.none(sql.addDapp, peer));
+				});
+			}
+		});
+
+        return t.batch(queries);
+    })
+    .then(function (data) {
+        library.logger.debug('Peers exported to database');
+        return setImmediate(cb);
+    })
+    .catch(function (err) {
+        library.logger.error('Export peers to database failed', {error: err.message || err});
+        return setImmediate(cb);
+    });
+};
+
 // Events
 Peers.prototype.onBind = function (scope) {
 	modules = scope;
 };
 
 Peers.prototype.onBlockchainReady = function () {
-	async.series({
+	steed.series({
 		insertSeeds: function (seriesCb) {
-			async.eachSeries(library.config.peers.list, function (peer, eachCb) {
+			steed.each(library.config.peers.list, function (peer, eachCb) {
 				self.update({
 					ip: peer.ip,
 					port: peer.port,
 					version: modules.system.getVersion(),
 					state: 2,
-					broadhash: modules.system.getBroadhash(),
+					// Apply our broadhash to seeds cause issues and sometimes will prevent sync
```

Comment:
`Apply our broadhash to seeds cause issues and sometimes will prevent sync` > `Apply our broadhash to seeds as can cause issues and sometimes will prevent sync`.